### PR TITLE
WIP: Software Architecture Improvement

### PR DESCRIPTION
**Pull request by guilyx**
- Created a Class outside of the main tree ( .h / .cpp )
- Added ReadMe, Licence and Gitignore ( ReadMe and Gitignore are work in progress )
- Should add vscode files to the gitignore
- Added a shell script supposed to install ROS Melodic without bugging
- Added videos of the demos ( navigation and mapping ) 

Do not merge yet ; still a work in progress

### CODE REVIEW

File: scripts/install_melodic.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	echo "ERROR : SHELL ENVIRONMENT NOT SUPPORTED (ZSH and BASH only !!)"
+fi
+
+sudo apt install python-rosinstall python-rosinstall-generator python-wstool build-essential
+
+mkdir -p ~/catkin_ws/src
+cd ~/catkin_ws/
+catkin_make
+
+source devel/setup.zsh
```
Comment (nicofilliol) [inline scripts/install_melodic.sh]: add echo "source /catkin_ws/devel/setup.bash" >> ~/.bashrc and same for zsh

File: src/RobotController.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+void RobotController::odomCallback(const nav_msgs::Odometry::ConstPtr& msg) {
+    // Read theta value of robot
+    float robot_theta = msg->pose.pose.orientation.w;
+}
+
+float RobotController::calculateGain(float value) {
+    float error = this->target_value - value;
+    float new_der_err = error - this->old_prop_error;
+    float new_int_err = integral_error + error;
```
Comment (nicofilliol) [inline src/RobotController.cpp]: this->integral_error

**Review by nicofilliol — COMMENTED** [review]
Good job

**Comment by guilyx** [comment]
Resoled conflicts, ready to merge

**Pull request merged**